pkg/cosign/tuf/client.go

```diff
@@ -503,49 +489,175 @@ func rootCacheDir() string {
 		if err != nil {
 			home = ""
 		}
-		return filepath.Join(home, ".sigstore", "root")
+		return filepath.FromSlash(filepath.Join(home, ".sigstore", "root"))
 	}
 	return rootDir
 }
 
 func cachedRemote(cacheRoot string) string {
-	return filepath.Join(cacheRoot, "remote.json")
+	return filepath.FromSlash(filepath.Join(cacheRoot, "remote.json"))
 }
 
 func cachedTargetsDir(cacheRoot string) string {
-	return filepath.Join(cacheRoot, "targets")
+	return filepath.FromSlash(filepath.Join(cacheRoot, "targets"))
 }
 
-type targetImpl interface {
-	Get(string) ([]byte, error)
-	setImpl
+// Local store implementations
+func newLocalStore() (client.LocalStore, error) {
+	if noCache() {
+		return client.MemoryLocalStore(), nil
+	}
+	tufDB := filepath.FromSlash(filepath.Join(rootCacheDir(), "tuf.db"))
+	local, err := tuf_leveldbstore.FileLocalStore(tufDB)
+	if err != nil {
+		return nil, fmt.Errorf("creating cached local store: %w", err)
+	}
+	return local, nil
 }
 
-type setImpl interface {
-	Set(string, []byte) error
+type localInitFunc func() (client.LocalStore, error)
+
+type embeddedLocalStore struct {
+	// This is where updated repository info will be written to, if needed.
+	writeable     client.LocalStore
+	writeableFunc localInitFunc
 }
 
-type memoryCache struct {
-	targets map[string][]byte
+var GetEmbedded = func() fs.FS {
```

Comment:
And then if we're going to do that, we may as well make it a parameter of the init method rather than something that needs to be mocked.

Reply:
Hmmmmm I don't want to lock it in because tests need to modify that on an individual basis

Otherwise I would need to pre-hook test runs, I think. But correct me if I'm wrong about that.

Reply:
Err, doesn't each test instantiate its own TUF client (via `NewFromEnv`)? You should be able to move the `GetEmbedded` call into `initializeTUF` or `wrapEmbedded`/`wrapEmbeddedLocal` just the first time. I also don't think it's that bad to have a separate constructor that *doesn't* access any environment variables that you could explicitly pass the fake FS into for tests.

Reply:
Oh!  I misunderstood what you meant by "init time" -- I thought that meant Golang "init" which happens once globally. 

I like the `initialize` idea.

Reply:
I think I'd rather pass this in to the constructor explicitly